src/open62541_compat.cpp

```diff
@@ -269,6 +268,8 @@ OpcUaType UaVariant::type() const
         return OpcUaType_String;
     else if(m_impl->type == &UA_TYPES[UA_TYPES_BOOLEAN])
     	return OpcUaType_Boolean;
+    else if(m_impl->type == &UA_TYPES[UA_TYPES_BYTESTRING])
+    	return OpcUaType_ByteString;
```

Comment:
Just a thought for an alternative implementation - we could fill a const static map with this mapping, then at 'normal' runtime we dig about in the map rather than dropping through the if/else. I think the boost::assign syntax is quite clean and concise for filling maps.